tests/merch.order.test.ts

```diff
@@ -0,0 +1,721 @@
+    const placedOrderResponse = await merchController.placeMerchOrder(placeMerchOrderRequest, member);
+
+    // cancel order
+    const { uuid } = placedOrderResponse.order;
+    expect(merchController.cancelMerchOrder({ uuid }, otherMember))
+      .rejects
+      .toThrow('Member cannot cancel other members\' orders');
+  });
+
+  test('admins can fulfill parts of a member\'s order', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake({ credits: 10000 });
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+    const affordableOption1 = MerchFactory.fakeOption({
+      quantity: 1,
+      price: 2000,
+      discountPercentage: 0,
+    });
+    const affordableOption2 = MerchFactory.fakeOption({
+      quantity: 1,
+      price: 3000,
+      discountPercentage: 0,
+    });
+    const pickupEvent = MerchFactory.fakeFutureOrderPickupEvent();
+
+    await new PortalState()
+      .createUsers(member)
+      .createMerchItemOption(affordableOption1)
+      .createMerchItemOption(affordableOption2)
+      .createOrderPickupEvents(pickupEvent)
+      .write();
+
+    const emailService = mock(EmailService);
+    when(emailService.sendOrderConfirmation(member.email, member.firstName, anything()))
+      .thenResolve();
+
+    // place order
+    const order = [
+      {
+        option: affordableOption1.uuid,
+        quantity: 1,
+      },
+      {
+        option: affordableOption2.uuid,
+        quantity: 1,
+      },
+    ];
+    const placeMerchOrderRequest = {
+      order,
+      pickupEvent: pickupEvent.uuid,
+    };
+    const merchController = ControllerFactory.merchStore(conn, instance(emailService));
+    const placedOrderResponse = await merchController.placeMerchOrder(placeMerchOrderRequest, member);
+
+    // fulfill a single item in an order
+    const placedOrder = placedOrderResponse.order;
+    const uuidParams = { uuid: placedOrder.uuid };
+    const fulfillMerchOrderItemsRequest = {
+      items: [
+        {
+          uuid: placedOrder.items[0].uuid,
+          fulfilled: true,
+        },
+        {
+          uuid: placedOrder.items[1].uuid,
+          fulfilled: false,
+        },
+      ],
+    };
+    await merchController.fulfillMerchOrderItems(uuidParams, fulfillMerchOrderItemsRequest, admin);
+
+    const getOrderResponse = await merchController.getOneMerchOrder(uuidParams, member);
+    const partiallyFulfilledOrder = getOrderResponse.order;
+
+    // order status should not be FULFILLED because entire order isn't fulfilled
+    expect(partiallyFulfilledOrder.status).toEqual(OrderStatus.PLACED);
+    expect(partiallyFulfilledOrder.items.some((item) => item.fulfilled)).toBeTruthy();
+    expect(partiallyFulfilledOrder.items.every((item) => item.fulfilled)).toBeFalsy();
+  });
+
+  test('admins can fulfill entire orders if every item of a member\'s order is fulfilled', async () => {
```

Comment:
Is the order correctly marked as fulfilled if the items are fulfilled in separate requests? Can we update this test to do that? I can't tell what'd happen just by looking at the service method (idk what happens in the loop where we fulfill order items when there's no update for an item since the map would presumably return null).

Reply:
Yeah I think `fulfillOrderItems` is a bit confusing in the way it's setup. Right now, we're requiring every `OrderItem` in an order to be passed to the controller, where the `fulfilled` boolean is either true or false, and if all item's `fulfilled = true` at the end, then the order is marked as fulfillled. I think this makes sense design wise since the UI would have a list of checkboxes, and the entire list can just be sent to the backend after the admin checks/unchecks fulfilled items. 

![image](https://user-images.githubusercontent.com/33337209/140656038-79514975-d99b-4fc0-b30d-dbfbd36890a2.png)

`fulfillOrderItems` has some extra complexity that was based on our old assumption that `OrderItem`s for different orders can be sent in the same request, and that not all `OrderItem`s are sent at the same time, so I'll update that method accordingly

Reply:
Update on the above - we actually are rolling with 'you can't unfulfill items' cause the above design would only display unfulfilled items. So `fulfillOrderItems` will take in only the items that it wants to fulfill, and users can fulfill their order across multiple requests. I'll add some more tests to test this.